library/std/src/sync/mod.rs

```diff
@@ -193,47 +203,42 @@ pub use self::once_lock::OnceLock;
 #[unstable(feature = "reentrant_lock", issue = "121440")]
 pub use self::reentrant_lock::{ReentrantLock, ReentrantLockGuard};
 
-// These make sense and exist only with poisoning.
+#[stable(feature = "rust1", since = "1.0.0")]
+#[expect(deprecated)]
+pub use self::once::ONCE_INIT;
+
+// Used for `sys::sync::once` implementations and `LazyLock` implementation.
+pub(crate) use self::once::OnceExclusiveState;
+
+// FIXME(sync_poison_mod): remove all `#[doc(inline)]` once the modules are stabilized.
+
+#[unstable(feature = "sync_poison_mod", issue = "134646")]
+pub mod poison;
+#[unstable(feature = "sync_nonpoison", issue = "134645")]
+pub mod nonpoison;
+
+// These exist only with poisoning.
 #[stable(feature = "rust1", since = "1.0.0")]
 #[doc(inline)]
 pub use self::poison::{LockResult, PoisonError};
 
-// These (should) exist in both flavors: with and without poisoning.
-// FIXME(sync_nonpoison): implement nonpoison versions:
-//  * Mutex (nonpoison_mutex)
-//  * Condvar (nonpoison_condvar)
-//  * Once (nonpoison_once)
-//  * RwLock (nonpoison_rwlock)
+// These exist in both flavors: with and without poisoning.
 // The historical default is the version with poisoning.
 #[stable(feature = "rust1", since = "1.0.0")]
 #[doc(inline)]
 pub use self::poison::{
-    Mutex, MutexGuard, TryLockError, TryLockResult,
+    TryLockError, TryLockResult,
+    Mutex, MutexGuard,
     Condvar,
-    Once, OnceState,
     RwLock, RwLockReadGuard, RwLockWriteGuard,
 };
```

Comment:
Looks like alphabetical ordering may have gotten flipped between `Mutex` and `TryLockError`. (would be nice if we could just remove the `rustfmt::skip` from this module but it does make things messy)